### DISCUSSION TITLE
`expect(() => {}).toEqual(() => {})`

### DISCUSSION THREAD

**Discussion by Jannchie** [vitest-dev/vitest#6495] [Q&A]
I want to determine whether two anonymous functions are completely identical. Is that possible? I tried `toEqual` but returns `false`.

**Comment by hi-ogawa** [comment]
Usually "completely identical" means javascript object identity and this would work:

```ts
const f = () => {};
expect(f).toBe(f)
```

However, I'm guessing this is not what you meant. Alternative would be to simply compare `toString()` like:

```ts
const f = () => {};
const g = () => {};
expect(f.toString()).toBe(g.toString())
```

Or if you want to compare two functions' outputs for the same input, then you need to generate input by yourselves. Something like fast-check https://github.com/dubzzz/fast-check would help this type of testing, for example:

```ts
const f = (x) => x + 1;
const g = (y) => 1 + y;

fc.assert(
  fc.property(fc.integer(), (x) => {
    expect(f(x)).toBe(g(x))
  }),
);
```
